db/db.go

```diff
@@ -105,31 +119,140 @@ func NewMinio(env e2e.Environment, name, bktName string, opts ...Option) *e2emon
 		command += "curl -sSL --tlsv1.3 -O 'https://raw.githubusercontent.com/minio/kes/master/root.key' -O 'https://raw.githubusercontent.com/minio/kes/master/root.cert' && cp root.* /home/me/ && "
 	}
 
+	var readiness e2e.ReadinessProbe
+
+	if o.minioOptions.enableTLS {
+		if err := os.MkdirAll(filepath.Join(f.Dir(), "certs", "CAs"), 0750); err != nil {
+			return &e2emon.InstrumentedRunnable{Runnable: e2e.NewFailedRunnable(name, errors.Wrap(err, "create certs dir"))}
+		}
+
+		if err := genCerts(
+			filepath.Join(f.Dir(), "certs", "public.crt"),
+			filepath.Join(f.Dir(), "certs", "private.key"),
+			filepath.Join(f.Dir(), "certs", "CAs", "ca.crt"),
+			fmt.Sprintf("%s-%s", env.Name(), name),
+		); err != nil {
+			return &e2emon.InstrumentedRunnable{Runnable: e2e.NewFailedRunnable(name, errors.Wrap(err, "fail to generate certs"))}
+		}
+
+		envVars = append(envVars, "ENABLE_HTTPS="+"1")
+		command = command + fmt.Sprintf(
+			"su - me -s /bin/sh -c 'mkdir -p %s && %s /opt/bin/minio server --certs-dir %s/certs --address :%v --quiet %v'",
+			filepath.Join(f.Dir(), bktName),
+			strings.Join(envVars, " "),
+			f.Dir(),
+			ports[AccessPortName],
+			f.Dir(),
+		)
+
+		readiness = e2e.NewHTTPSReadinessProbe(
+			AccessPortName,
+			"/minio/health/cluster",
+			200,
+			200,
+		)
+	} else {
+		envVars = append(envVars, "ENABLE_HTTPS="+"0")
+		command = command + fmt.Sprintf(
```

Comment:
Nit: we could have one `command` var instead of two separate in each branch and just (not) include `--certs-dir` depending on HTTPS option